features/285-design-webhooks/design.md

```diff
@@ -1,129 +1,324 @@
-# next webhooks design
+# Background
 
-## Overview
+Web-hooks would be used to notify external services of events in auth gear.
 
-skygear sends webhooks for events that happen in your app, and are designed to help a developer to monitor events, sync database, and many other use cases.
 
-When one of events is triggered, skygear sends a HTTP POST with a payload to the webhook's configured URL.
+# Proposed Designs
 
-## SYNC and ASYNC hooks
+## Events
```

Comment:
Mention all kinds of hooks here. BEFORE, AFTER and USER_SYNC. This section is the very first part of this document so it should be an overview.